### PR TITLE
R style columns

### PR DESCRIPTION
**Pull request by gidden**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

Who else has gotten very useful data from very useful colleages in that very pesky R-style data format, where year columns look like `X2015` instead of just `2015`? Worry no more, `pyam` has your back!

### CODE REVIEW

**Comment by gidden** [comment]
Mind taking a look @znicholls? I want to give @danielhuppmann a bit of a break from reviews (at least for one!).

File: pyam/utils.py [hunk 1]
```diff
@@ -132,6 +132,24 @@ def format_data(df, **kwargs):
...
+                    #  bingo! was X2015 R-style, return the integer
+                    return int(second)
+                except:
+                    # nope, not an int, fall down to final return statement
+                    pass
+        except:
+            # not a string/iterable/etc, fall down to final return statement
+            pass
+        return c
+    df.columns = map(convert_r_columns, df.columns)
```
Comment (znicholls) [inline pyam/utils.py]:
```suggestion
    df.columns = df.columns.map(convert_r_columns)
```

may be more pandas like?

**Comment by gidden** [comment]
Ok, can do. Errors in CI are not related, FYI.

**Comment by znicholls** [comment]
> Errors in CI are not related, FYI.

I guessed as much. Do you want me to split my fix from #198 out into a separate PR to see if that helps?

**Comment by znicholls** [comment]
> Do you want me to split my fix from #198 out into a separate PR to see if that helps?

Ignore that, just looked at the errors properly

File: pyam/utils.py [hunk 2]
```diff
@@ -132,6 +132,24 @@ def format_data(df, **kwargs):
     if isinstance(df, pd.Series):
         df = df.to_frame()
 
+    # Check for R-style year columns, converting where necessary
```
Comment (danielhuppmann) [inline pyam/utils.py]: suggest to move that function out of `format_data()` - the function is already too long...

Comment (gidden) [reply pyam/utils.py]: But it is also the catch-all function for saying "here is an input data set, make it into a `pyam.IamDataFrame`. I agree we could break `format_data` into multiple functions that all get called, but I don't think that's in the scope of this PR. Happy to make an issue though!

**Comment by gidden** [comment]
tests will fail again because our backends are down from public access..

**Comment by gidden** [comment]
should be good to go now

**Comment by gidden** [comment]
once issue discussion resolved with @danielhuppmann

**Pull request merged**